### ISSUE TITLE
When opening a context manager, prefer `await nursery.start()` over `nursery.start_soon()`

### ISSUE THREAD

**Issue by Zac-HD** [Zac-HD/flake8-trio#57]
One pattern I've seen a few times lately is a context manager which ensures some external service, e.g. an external process to interact with, is available.  The code might look something like this:

```python
@asynccontextmanager
def run_sampling_profiler():
    async with trio.open_nursery() as nursery:
        nursery.start_soon(trio.run_process, ["start-sampling-profiler", str(os.getpid())])
        yield
```

The only problem is that the code in the calling `with`-block can start running before our external process is actually running!  To fix this, we should have used `await nursery.start(trio.run_process, ...)`.

So to *lint* for this, we should check `nursery.start_soon()` calls either inside `async def __aenter__(self):` or before the `yield` statement in a function like the above, and raise a warning if the target callable could instead be used with `nursery.start()`.  But how do we distinguish those startable targets?

- They [accept a `task_status=` keyword argument](https://trio.readthedocs.io/en/stable/reference-core.html#trio.Nursery.start)... but doing cross-module introspection would be a pretty nasty latency hit
- We can list out known cases - [search `task_status=` on this page](https://trio.readthedocs.io/en/stable/reference-io.html), and I'll be happy to expand the list if there are other suggestions
- We can support a configuration option to add more on a project-specific basis
